### DISCUSSION TITLE
Any idea to encapsulate a session fearure just like in python request?

### DISCUSSION THREAD

**Discussion by ZhechenLi** [sindresorhus/got#2058] [Q&A] [answered]
#### What problem are you trying to solve?
Any idea to encapsulate a session fearure just like in python request to management cookie automatically?
...

#### Describe the feature
We're a team which make some crawler job, and some of then is transfer from python.
In python there are have a session feature which can persist certain parameters across requests like cookie.

```python
s = requests.Session()

s.get('https://httpbin.org/cookies/set/sessioncookie/123456789')
r = s.get('https://httpbin.org/cookies')

print(r.text)
# '{"cookies": {"sessioncookie": "123456789"}}'
```
you can find more tips from [this link](https://requests.readthedocs.io/en/master/user/advanced/).

...

**Comment by sindresorhus** [comment, marked as answer]
Cookie handling is documented here: https://github.com/sindresorhus/got/blob/main/documentation/tips.md#cookies
